### PR TITLE
Added a simple hash wrapper

### PR DESCRIPTION
**Pull request by rullzer**
Lets see if my ICE hacking works

### CODE REVIEW

File: src/HashWrapper.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	 */
+	public static function wrap($source, $hash, $callback) {
+		$context = stream_context_create(array(
+			'hash' => array(
+				'source' => $source,
+				'hash' => $hash,
+				'callback' => $callback,
+			)
+		));
+		return Wrapper::wrapSource($source, $context, 'callback', '\Icewind\Streams\SeekableWrapper');
```
Comment (icewind1991) [inline src/HashWrapper.php]: you need to set the correct classname.

I'll see if I can update the base class to remove the need to specify it in the first place

**Comment by icewind1991** [comment]
and please rebase to get properly working ci

File: src/HashWrapper.php (hunk 1 above)
Comment (icewind1991) [reply src/HashWrapper.php]: with master you no longer need to specify the last 2 arguments, as long as you change `Wrapper::wrapSource` to `self::wrapSource`

**Comment by rullzer** [comment]
Done

**Comment by icewind1991** [comment]
I'll rename it to `ReadHashWrapper` though and add a write equivalent

**Pull request merged**